packages/ui/src/ui/pages/system/Masters/Instance.tsx

```diff
@@ -12,110 +11,96 @@ import ClipboardButton from '../../../components/ClipboardButton/ClipboardButton
 import {ChangeMaintenanceButton} from './ChangeMaintenanceButton';
 import block from 'bem-cn-lite';
 import {MasterInstance} from '../../../store/selectors/system/masters';
-import ypath from '../../../common/thor/ypath';
 import './MasterGroup.scss';
 
 const b = block('master-group');
 
-export class Instance extends Component {
-    static instanceStateToTheme = {
-        active: 'online',
-        connected: 'online',
-        leading: 'online',
-        disconnected: 'banned',
-        following: 'following',
-        follower_recovery: 'banned',
-        leader_recovery: 'banned',
-        standby: 'banned',
-        elections: 'offline',
-        stopped: 'offline',
-        unknown: 'offline',
-        online: 'online',
-        offline: 'offline',
-    };
+const instanceStateToTheme: Record<string, NodeQuadTheme> = {
+    active: 'online',
+    connected: 'online',
+    leading: 'online',
+    disconnected: 'banned',
+    following: 'following',
+    follower_recovery: 'banned',
+    leader_recovery: 'banned',
+    standby: 'banned',
+    elections: 'offline',
+    stopped: 'offline',
+    unknown: 'offline',
+    online: 'online',
+    offline: 'offline',
+};
 
-    static propTypes = {
-        instance: MasterInstance,
-        hostType: PropTypes.string.isRequired,
-        allowVoting: PropTypes.bool,
-        allowService: PropTypes.bool,
-    };
+type Props = {
+    instance: MasterInstance;
+    hostType: string;
+    allowVoting?: boolean;
+    allowService?: boolean;
+};
 
-    render() {
-        const {instance, hostType, allowVoting, allowService} = this.props;
-        const {state, $address, $physicalAddress, $attributes, $rowAddress} = instance;
-        const address = hostType === 'host' ? $address : $physicalAddress;
-        const maintenance = ypath.getValue($rowAddress, '/attributes/maintenance');
-        const maintenanceMessage = maintenance
-            ? ypath.getValue($rowAddress, '/attributes/maintenance_message') || 'Maintenance'
-            : '';
-        const {voting} = $attributes ?? {};
-        // do not use `!voting` cause `voting === undefined` is the same as `voting === true`
-        const denyVoting = allowVoting && voting === false;
-        const theme =
-            denyVoting && state === 'following'
-                ? 'nonvoting'
-                : Instance.instanceStateToTheme[state];
-        const addressWithoutPort = hammer.format['Address'](address);
+export const Instance: FC<Props> = ({instance, hostType, allowVoting, allowService}) => {
+    const {state, $address, $physicalAddress} = instance;
+    const attributes = instance.$attributes as Record<string, any>;
```

Comment:
Is it possible to provide more details for the type (instead of `Record<string, any>`)?
(also we have helper type `CypressNode`)